puppet/spec/unit/indirector/facts/puppetdb_spec.rb

```diff
@@ -168,10 +169,14 @@ def find_facts()
       body = [{"certname" => "some_node", "environment" => "production", "name" => "a", "value" => "1"},
               {"certname" => "some_node", "environment" => "production", "name" => "b", "value" => "2"}].to_json
 
-      response = Net::HTTPOK.new('1.1', 200, 'OK')
+      response = Puppet::HTTP::Response.new(Net::HTTPOK.new('1.1', 200, 'OK'), "mock url")
       response.stubs(:body).returns body
 
-      http.stubs(:get).with("/pdb/query/v4/nodes/some_node/facts", subject.headers, options).returns response
+      http.stubs(:get).with do |uri, opts|
+        "/pdb/query/v4/nodes/some_node/facts" == uri.path &&
```

Comment:
Does this need to be `expect("/pdb/query/v4/nodes/some_node/facts").to eq(uri.path)` etc?

Reply:
Our tests are still using Mocha I believe, so the yield needs to return a boolean https://mocha.jamesmead.org/Mocha/Expectation.html#with-instance_method

Reply:
Ahhh yeah good ol' mocha